Makefiles/Makefile.docker

```diff
@@ -8,41 +8,38 @@ ifeq ($(ARCH),x86_64)
 	ARCH := x64
 endif
 
+# Use to get one commit by hash, as opposed to a branch or tag name
+clone_hash = mkdir -p $(1) && cd $(1) && git init -q . && git remote add origin $(2) && git fetch -q --depth 1 origin $(3) && git reset -q --hard $(3)
+# Use to go to a specific commit by hash while still getting the entire repo, in case a build process operates on the git history
+clone_all_to_hash = mkdir -p $(1) && cd $(1) && git clone -q $(2) . && git checkout -q $(3)
+
 GN_ARGS = is_component_build=false v8_use_external_startup_data=false v8_enable_i18n_support=false is_clang=false use_gold=false use_custom_libcxx=false treat_warnings_as_errors=false v8_monolithic=true
 
 
 build/gn/out/gn:
 	@echo "Setting up build directory"
 	@mkdir -p build
 	@echo "Downloading gn source"
-	@git clone -q https://gn.googlesource.com/gn.git build/gn
-	@cd build/gn && git checkout -q eea3906f0e2a8d3622080127d2005ff214d51383
+	@$(call clone_all_to_hash,build/gn,https://gn.googlesource.com/gn.git,eea3906f0e2a8d3622080127d2005ff214d51383)
```

Comment:
I'm not a big fan of this.  I understand the DRY aspect of it, but it really makes it a lot harder to quickly understand exactly what steps are being done in the download and build process.

if this could be unrolled, I'd be much happier.

Reply:
Consider it unrolled. 🙂

Reply:
awesome, thanks!